sapi/embed/config.m4

```diff
@@ -22,6 +45,46 @@ if test "$PHP_EMBED" != "no"; then
       PHP_EMBED_TYPE=static
       INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)\$(orig_libdir); \$(INSTALL) -m 0644 $SAPI_STATIC \$(INSTALL_ROOT)\$(orig_libdir)"
     ],
+    [main], [
+      AC_MSG_RESULT([main])
+      AC_PATH_PROG([XXD], [xxd])
+      if test ! -x "$XXD"; then
+        AC_MSG_ERROR([xxd not found, required for --with-embed-main])
+      fi
+      AC_MSG_CHECKING([for access to $PHP_EMBED_MAIN])
+      if test ! -f "$PHP_EMBED_MAIN"; then
+        AC_MSG_ERROR([--with-embed-main argument "$PHP_EMBED_MAIN" not found])
+      else
+        AC_MSG_RESULT([ok])
+      fi
+
+      AC_MSG_CHECKING([generating sapi/embed/php_embed_main.h])
+      $XXD -i -n php_embed_main "$PHP_EMBED_MAIN" sapi/embed/php_embed_main.h
+      if test $? -ne 0; then
+        AC_MSG_ERROR([failed])
+      else
+        AC_MSG_RESULT([ok])
+      fi
+
+      AC_MSG_CHECKING([proceeding to final configure of embed])
+      LIBPHP_CFLAGS="-static"
+      PHP_EMBED_TYPE=static
+      INSTALL_IT="\
+        \$(mkinstalldirs) \$(INSTALL_ROOT)\$(orig_libdir); \
+        \$(INSTALL) -m 0644 \$(SAPI_STATIC) \$(INSTALL_ROOT)\$(orig_libdir); \
+        \$(mkinstalldirs) \$(INSTALL_ROOT)\$(bindir); \
+        \$(LIBTOOL) --tag=CC --mode=compile \$(CC) \
+          \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(INCLUDES) -DHAVE_EMBED_MAIN \
+            -c sapi/embed/php_embed.c -o sapi/embed/php_embed_main.lo; \
+        \$(LIBTOOL) --tag=CC --mode=link \$(CC) \
+          -export-dynamic \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \
+            sapi/embed/php_embed_main.lo \
+            \$(PHP_GLOBAL_OBJS) \$(PHP_SAPI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) \
+          -o \$(INSTALL_ROOT)\$(bindir)/php-embed-main"
+        if test "$PHP_EMBED_LINK" != "no"; then
+          AC_DEFINE_UNQUOTED([PHP_EMBED_LINK], ["$PHP_EMBED_LINK"], [PHP_EMBED_LINK])
```

Comment:
if this is yes/empty we could use basename of `PHP_EMBED_MAIN`, this will normally be correct.